### PR TITLE
Do not save clientId in APIKEY subscription

### PR DESCRIPTION
**Pull request by phiz71**
## Issue

gravitee-io/issues#8855
https://gravitee.atlassian.net/browse/APIM-676

## Description

APIM gateway has a subscription cache that is initialized at startup, updated on API deployment and with a scheduler looking in database.
In this cache, we put all subscriptions with a clientId.

When a call is made on a JWT or OAuth plan, the gateway looks into this cache to find if a subscription exists for a given API and a clientID (found in the bearer token).

As APIKey subscriptions are also created with a clientId, they are mixed with JWT and OAuth2 subscriptions and the selection of the subscription in the cache fails.

This PR:
 - change the save and update feature to avoid saving clientID in APIKEY subscription
 - create an upgrader to clean existing subscriptions

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-snammpccpu.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-676-subscirption-lost-on-redeploy/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/InstallationService.java [hunk 1]
```diff
@@ -29,6 +29,7 @@ public interface InstallationService {
     String PLANS_DATA_UPGRADER_STATUS = "PLANS_DATA_UPGRADER_V2_STATUS";
     String APPLICATION_API_KEY_MODE_UPGRADER_STATUS = "API_KEY_MODE_UPGRADER_STATUS";
     String API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS = "API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS";
+    String CLIENT_ID_IN_API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS = "CLIENBT_ID_IN_API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS";
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/InstallationService.java]:
```suggestion
    String CLIENT_ID_IN_API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS = "CLIENT_ID_IN_API_KEY_SUBSCRIPTIONS_UPGRADER_STATUS";
```

Comment (phiz71) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/InstallationService.java]: 🤦🏻‍♂️ good catch 😅

**Comment by phiz71** [comment]
Tested with 60000 APIKEY subscriptions, all with a clientId.
It took about 2min to upgrade all the subscriptions at startup, in the upgrader phase.

**Review by marcambier — APPROVED** [review]
:+1:

**Pull request merged**